src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java

```diff
@@ -186,6 +202,19 @@ public void onTerminateR()
             "\n\n" +
             "Are you sure you want to terminate R?");
    }
+
+   @Override
+   public void onConsoleBusy(ConsoleBusyEvent event)
+   {
+      isBusy_ = event.isBusy();
+
+      if (!isBusy_ && pendingInterrupt_)
+      {
+         eventBus_.fireEvent(new InterruptStatusEvent(
+               InterruptStatusEvent.INTERRUPT_COMPLETED));
+         finishInterrupt(pendingInterruptHandler_);
```

Comment:
Should we reset `pendingInterrupt_` + `pendingInterruptHandler_` here (after invoking `finishInterrupt()`)?